### ISSUE TITLE
JX Eval Leak

### ISSUE THREAD

**Issue by dthain** [cooperative-computing-lab/cctools#2349]
JX has a memory leak on this expression.
Haven't been able to quite track it down.
Related to a function generating an array as a result?

```
[dthain@cclws15 src]$ valgrind --leak-check=full ./jx_test
==26585== Memcheck, a memory error detector
==26585== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==26585== Using Valgrind-3.12.0 and LibVEX; rerun with -h for copyright info
==26585== Command: ./jx_test
==26585== 
Enter context expression (or {} for an empty context):
{};
Now enter expressions:
range(1,3);
expression: range(1,3)
value:      [1,2]

==26585== 
==26585== HEAP SUMMARY:
==26585==     in use at exit: 160 bytes in 5 blocks
==26585==   total heap usage: 20 allocs, 15 frees, 66,206 bytes allocated
==26585== 
==26585== 160 (32 direct, 128 indirect) bytes in 1 blocks are definitely lost in loss record 5 of 5
==26585==    at 0x4C2B975: calloc (vg_replace_malloc.c:711)
==26585==    by 0x413174: xxcalloc (xxmalloc.c:51)
==26585==    by 0x402C14: jx_create (jx.c:48)
==26585==    by 0x402F0B: jx_array (jx.c:129)
==26585==    by 0x40AA6F: jx_eval (jx_eval.c:625)
==26585==    by 0x409BD3: jx_eval_operator (jx_eval.c:360)
==26585==    by 0x40AAC1: jx_eval (jx_eval.c:631)
==26585==    by 0x402978: main (jx_test.c:50)
==26585== 
==26585== LEAK SUMMARY:
==26585==    definitely lost: 32 bytes in 1 blocks
==26585==    indirectly lost: 128 bytes in 4 blocks
==26585==      possibly lost: 0 bytes in 0 blocks
==26585==    still reachable: 0 bytes in 0 blocks
==26585==         suppressed: 0 bytes in 0 blocks
==26585== 
==26585== For counts of detected and suppressed errors, rerun with: -v
==26585== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

**Comment by trshaffer** [comment]
I think this is fixed in #2342. I tracked down several memory leaks in eval and parse

**Comment by dthain** [comment]
Excellent, thanks.

**Comment by trshaffer** [comment]
Fixed in e0bd59a35dd003f466cd3824d68b1aa5e5926aee, closing
